chaoslib/openebs/cstor_pool_failure.yaml

```diff
@@ -0,0 +1,73 @@
+- name: Derive PV from application PVC 
+  shell: >
+    kubectl get pvc {{ app_pvc }}
+    -o custom-columns=:spec.volumeName -n {{ app_ns }}
+    --no-headers
+  args:
+    executable: /bin/bash
+  register: pv
+
+- name: Derive SC from application PVC 
+  shell: >
+    kubectl get pvc {{ app_pvc }}
+    -o custom-columns=:spec.storageClassName -n {{ app_ns }}
+    --no-headers
+  args:
+    executable: /bin/bash
+  register: sc
+
+- name: Derive ReplicaCount from SC
+  shell: >
+    kubectl get sc {{ sc.stdout }} -n {{ operator_ns }} -o json |
+    jq '.metadata.annotations."cas.openebs.io/config"' -r |
+    grep -A 1 -w "name: ReplicaCount" | grep -w value | awk '{print $2}'
+  args:
+    executable: /bin/bash
+  register: replicacount
+
+- name: Set default value for replicacount if it is empty
+  set_fact:
+    replicacnt: "3"
+  when: "replicacount.stdout == \"\""
+
+- name: Set default value for replicacount if it is non-empty
+  set_fact:
+    replicacnt: "{{ replicacount.stdout | int }}"
+  when: "replicacount.stdout != \"\""
+
+- name: Get CVR count from pv
+  shell: >
+    kubectl get cvr -n {{ operator_ns }}
+    -l openebs.io/persistent-volume={{ pv.stdout }} --no-headers -o json |
+    jq '.items | length'
+  args:
+    executable: /bin/bash
+  register: cvr_count
+
+- name: Compare ReplicaCount and cvr_count to verify provisioning
+  debug:
+    msg:
+      - "replicaCount: {{ replicacount.stdout }}"
+      - "replicaCountInt: {{ replicacount.stdout | int }}"
+      - "replicacnt: {{ replicacnt }}"
+      - "cvr_count: {{ cvr_count.stdout | int }}"
+  failed_when: "replicacnt|int != cvr_count.stdout|int"
+
+- name: Get CVR status list from pv
+  shell: >
+    kubectl get cvr -n {{ operator_ns }}
+    -l openebs.io/persistent-volume={{ pv.stdout }} --no-headers -o json |
+    jq '.items[].status.phase' -r
+  args:
+    executable: /bin/bash
+  register: cvr_status_phase
+
+- name: Check status of cvr
+  command: echo "{{ item }}"
+  failed_when: "item != \"Online\""
+  with_items:
+    - "{{ cvr_status_phase.stdout_lines }}"
+
+- include: cstor_delete_pool_deployment.yaml
+
+- include: cstor_delete_pool_deployment.yaml
```

Comment:
A good test to do it more than once, as shuf will ensure selection of a different cvr (and thereby pool) most times. Would it be good to define the chaos iteration count in the test_vars & also include a volume sanity check before each iteration of chaos - so we will fail faster, if at all.

Reply:
shuf can give the same cvr also, but, that should be fine for us even if it is different.
This same file verifies that deployment is started again.